### ISSUE TITLE
a question about loss function

### ISSUE THREAD

**Issue by nswei** [YunYang1994/tensorflow-yolov3#81]
In loss_layer function(), the parameters 'feature_map_i' and 'y_true' are of a batch'.
In 'valid_true_boxes = tf.boolean_mask(y_true[..., 0:4], tf.cast(object_mask[..., 0], 'bool'))', you get a [v, 4] tensor.
The 'v' is the number of gtboxs in a whole batch, and then you calculate iou between the gtboxes and featuremaps,the shape of which is '[N, 13, 13, 3, 4]'.
You get a '[N, 13, 13, 3, V]' tensor, which means that you match the gtboxes of a whole batch with the prediction of a single image of the batch.
As far as I am concerned, you should match the gtboxes of an image with the prediction of an image.
What do you think of it?
